server/spike/adk/spike_05_steering.py

```
@@ -0,0 +1,156 @@
+steering_applied = False
+
+
+def before_tool_callback(tool, args, tool_context):
+    """Intercept tool calls and steer agent behavior."""
+    global steering_applied
+
+    # After the first research call, we want to steer the agent
+    # to focus on a specific aspect
+    if tool.name == "research_topic" and len(research_topics) >= 1 and not steering_applied:
+        steering_applied = True
+        print(f"  [STEERING] Intercepted research_topic call. Modifying args to focus on pricing.")
```

Ruff (0.15.10)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)